test/caption-stream.test.js

```diff
@@ -242,6 +245,104 @@ QUnit.test('sorting is fun', function() {
+    { pts: 11000, type: 0, ccData: 0x1c2d },
+    { pts: 12000, type: 0, ccData: 0x1425 },
+    { pts: 13000, type: 0, ccData: characters('1b') },
+    { pts: 14000, type: 0, ccData: characters('1c') },
+    { pts: 15000, type: 0, ccData: 0x142d },
+    { pts: 16000, type: 1, ccData: 0x152d },
+    { pts: 17000, type: 1, ccData: 0x1d2d },
+    { pts: 18000, type: 0, ccData: characters('2b') },
+    { pts: 19000, type: 0, ccData: 0x1c2d }
+  ];
+
+  var seiNals = packets.map(makeSeiFromCaptionPacket);
+  seiNals.forEach(captionStream.push, captionStream);
+  captionStream.flush();
+
+  QUnit.equal(captions.length, 6, 'got all captions');
+  QUnit.equal(captions[0].text, '1a', 'cc1 first row');
+  QUnit.equal(captions[1].text, '2a', 'cc2 first row');
+  QUnit.equal(captions[2].text, '1a\n1b1c', 'cc1 first and second row');
+  QUnit.equal(captions[3].text, '3a', 'cc3 first row');
+  QUnit.equal(captions[4].text, '4a4b', 'cc4 first row');
+  QUnit.equal(captions[5].text, '2a\n2b', 'cc2 first and second row');
+
+});
+
+QUnit.test('drops data until first command that sets activeChannel', function() {
+  var captions = [];
+  captionStream.ccStreams_.forEach(function(cc) {
+    cc.on('data', function(caption) {
+      captions.push(caption);
+    });
+  });
+
+  var packets = [
+    { pts: 0 * 1000, ccData: characters('no'), type: 0 },
+    { pts: 0 * 1000, ccData: characters('t '), type: 0 },
+    { pts: 0 * 1000, ccData: characters('th'), type: 0 },
+    { pts: 0 * 1000, ccData: characters('is'), type: 0 },
+    { pts: 1 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 1 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 2 * 1000, ccData: 0x142f, type: 0 },
+    // RCL, resume caption loading
+    { pts: 3 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 3 * 1000, ccData: 0x142e, type: 0 },
+    { pts: 4 * 1000, ccData: characters('te'), type: 0 },
+    { pts: 4 * 1000, ccData: characters('st'), type: 0 },
+    { pts: 5 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 5 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 6 * 1000, ccData: 0x142f, type: 0 }
+  ];
+
+  var seiNals = packets.map(makeSeiFromCaptionPacket);
+  seiNals.forEach(captionStream.push, captionStream);
+  captionStream.flush();
+
+  QUnit.equal(captions.length, 1, 'caption 1 dropped');
+  QUnit.equal(captions[0].text, 'test', 'caption with ambiguous channel dropped');
+  QUnit.equal(captions[0].stream, 'CC1', 'caption went to right channel');
+});
+
+QUnit.test('ignores CEA708 captions', function() {
+  var captions = [];
+  captionStream.ccStreams_.forEach(function(cc) {
+    cc.on('data', function(caption) {
+      captions.push(caption);
+    });
+  });
+
+  var seiNals = multichannelCaptions.map(makeSeiFromCaptionPacket);
+  seiNals.forEach(captionStream.push, captionStream);
+  captionStream.flush();
+
+  QUnit.equal(captions.length, 3, 'parsed three captions');
+  QUnit.equal(captions[0].text, 'BUT IT\'S NOT SUFFERING\nRIGHW.', 'parsed first caption correctly');
```

Comment:
Yup. It was like that when I got here :innocent:

It comes from an external file and I didn't feel like reprocessing the caption by hand.